src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx

```diff
@@ -79,6 +79,9 @@
   <data name="Arg_CryptographyException" xml:space="preserve">
     <value>Error occurred during a cryptographic operation.</value>
   </data>
+  <data name="Cryptography_BadData" xml:space="preserve">
+    <value>Bad Data.</value>
```

Comment:
This is actually the message from Windows for NTE_BAD_DATA, so macOS and Windows will have the same e.Message value in this case.  Does that change your opinion? :smile: